app/models/concerns/hostext/search.rb

```diff
@@ -41,6 +41,7 @@ module Search
       scoped_search :in => :architecture, :on => :name,    :complete_value => true, :rename => :architecture
       scoped_search :in => :puppet_proxy, :on => :name,    :complete_value => true, :rename => :puppetmaster
       scoped_search :in => :puppet_ca_proxy, :on => :name,    :complete_value => true, :rename => :puppet_ca
+      scoped_search :in => :puppet_proxy, :on => :name,    :complete_value => true, :rename => :smart_proxy, :ext_method => :search_by_proxy
```

Comment:
Probably we need to set it to explicit search only, especially when you allow `complete_value`.